--- Base.pm	2004/12/10 10:10:33	1.1
+++ Base.pm	2004/12/10 10:16:55
@@ -54,13 +54,19 @@
   return $self;
 }
 
+
+my $Package_Num = 0;
 sub new_from_context {
   my ($package, %args) = @_;
   
   # XXX Read the META.yml and see whether we need to run the Build.PL
   
-  # Run the Build.PL
-  $package->run_perl_script('Build.PL');
+  # Run each Build.PL in its own package to avoid conflicts.
+  eval sprintf q{
+      package %s;
+      do './Build.PL';
+  }, "Module::Build::Build_PL::PACK".$Package_Num++;
+
   my $self = $package->resume;
   $self->merge_args(undef, %args);
   return $self;
